lib/logstash/plugin.rb

```diff
@@ -121,9 +121,22 @@ def inspect
     end
   end
 
+  public
+  def self.auto_download?
+    @auto_download
+  end
+
+  public
+  def self.auto_download=(bool)
+    @auto_download = bool
+  end
+
   # Look up a plugin by type and name.
   public
   def self.lookup(type, name)
+
+    retried_once ||= false
```

Comment:
Lookup is not that bad, isn't? 

if renamed we should add the question mark at end, so would like ``` downloaded? ``` for me better than ```auto_```.